packages/playwright-core/src/server/fetch.ts

```diff
@@ -336,12 +336,14 @@ export abstract class APIRequestContext extends SdkObject {
             redirectOptions.rejectUnauthorized = false;
 
           // HTTP-redirect fetch step 4: If locationURL is null, then return response.
-          if (response.headers.location) {
+          // Best-effort UTF-8 decoding, per spec it's US-ASCII only, but browsers are more lenient.
+          const locationHeaderValue = Buffer.from(response.headers.location ?? '', 'binary').toString('utf8');
```

Comment:
response.headers.location is already a string, which encoding is it in?

Reply:
Let's change it to `latin1` and add a comment that it is constructed as a one byte in v8 via [NewFromOneByte](https://source.chromium.org/chromium/chromium/src/+/main:v8/include/v8-primitive.h;bpv=1;bpt=1;l=432?q=v8::String::NewFromOneByte&ss=chromium%2Fchromium%2Fsrc&gsn=NewFromOneByte&gs=KYTHE%3A%2F%2FKrQCCpcBa3l0aGU6Ly9jaHJvbWl1bS5nb29nbGVzb3VyY2UuY29tL2NvZGVzZWFyY2gvY2hyb21pdW0vc3JjLy9tYWluP2xhbmc9YyUyQiUyQj9wYXRoPXY4L2luY2x1ZGUvdjgtcHJpbWl0aXZlLmgjM1ViNFRWNXdyQmRvYU1pUkRNVWc2LUFIdk9ILTN1THExUVBCcnF0T1BRMAqXAWt5dGhlOi8vY2hyb21pdW0uZ29vZ2xlc291cmNlLmNvbS9jb2Rlc2VhcmNoL2Nocm9taXVtL3NyYy8vbWFpbj9sYW5nPWMlMkIlMkI_cGF0aD12OC9pbmNsdWRlL3Y4LXByaW1pdGl2ZS5oI2V3SHBHX1lfVXRKOXJ4YzN0eVo0aXlxSXdmMFJ6d1ROOGVjYnczWVV3Mmc%3D)(